keras_hub/src/models/gemma3/gemma3_causal_lm.py

```diff
@@ -0,0 +1,325 @@
+    default, `"greedy"` sampling will be used.
+
+    This model can optionally be configured with a `preprocessor` layer, in
+    which case it will automatically apply preprocessing to string inputs during
+    `fit()`, `predict()`, `evaluate()` and `generate()`. This is done by default
+    when creating the model with `from_preset()`.
+
+    Args:
+        backbone: A `keras_hub.models.Gemma3Backbone` instance.
+        preprocessor: A `keras_hub.models.Gemma3CausalLMPreprocessor` or
+            `None`. If `None`, this model will not apply preprocessing, and
+            inputs should be preprocessed before calling the model.
+
+    Examples:
+
+    Use `generate()` to do text generation.
+    ```python
+    image = np.random.rand(224, 224, 3)
+    gemma3_lm = keras_hub.models.Gemma3CausalLM.from_preset(
+        "gemma3_3b_mix_224"
+    )
+    gemma3_lm.generate(
+      {
+        "images": image,
+        "text": ["answer en where is the cow standing?\\n"]
+      }
+    )
+
+    # Generate with batched prompts.
+    gemma3_lm.generate(
+      {
+        "images": [image, image],
+        "text": ["answer en where is the cow standing?\\n", "caption en\\n"]
+      }
+    )
+    ```
+
+    Use `generate()` without preprocessing.
+    ```python
+    image = np.random.rand(224, 224, 3)
+    inputs = {
+        "images": [image, image],
+        # Token ids for "<bos> Keras is".
+        "token_ids": np.array([[2, 214064, 603, 0, 0, 0, 0]] * 2),
+        # Use `"padding_mask"` to indicate values that should not be overridden.
+        "padding_mask": np.array([[1, 1, 1, 0, 0, 0, 0]] * 2),
+    }
+
+    gemma3_lm = keras_hub.models.Gemma3CausalLM.from_preset(
+        "gemma3_3b_mix_224",
+        preprocessor=None,
+    )
+    gemma3_lm.generate(inputs)
+    ```
+
+    Custom backbone and vocabulary.
+    ```python
+    tokenizer = keras_hub.models.Gemma3Tokenizer(
+        proto="proto.spm",
+    )
+    preprocessor = keras_hub.models.Gemma3CausalLMPreprocessor(
+        tokenizer=tokenizer,
+        sequence_length=128,
+    )
+    backbone = keras_hub.models.Gemma3Backbone()
+    gemma3_lm = keras_hub.models.Gemma3CausalLM(
+        backbone=backbone,
+        preprocessor=preprocessor,
+    )
+    ```
+    """
+
+    backbone_cls = Gemma3Backbone
+    preprocessor_cls = Gemma3CausalLMPreprocessor
+
+    def __init__(
+        self,
+        preprocessor,
+        backbone,
+        **kwargs,
+    ):
+        # === Layers ===
+        self.preprocessor = preprocessor
+        self.backbone = backbone
+
+        # === Functional Model ===
+        # This must be "backbone.input" i.e. the full input structure,
+        # rather than "backbone.inputs" which is the flattened list of inputs.
+        inputs = backbone.input
+        hidden_state = backbone(inputs=inputs)
+        outputs = backbone.token_embedding(hidden_state, reverse=True)
+        # outputs = outputs[:, backbone.image_sequence_length :, :]
```

Comment:
I think we should delete this line for sure right? It doesn't make sense.

Reply:
Yeah, needs to be deleted.